lib/libvmod_blob/hex.c

```diff
@@ -133,13 +146,15 @@ hex_decode(const enum encoding dec, char *restrict const buf,
 
 	if (len == 0)
 		return (0);
-	if (n != -1 && len > n)
-		len = n;
+	if (inlen != -1 && len > inlen)
+		len = inlen;
 
-	if (((len+1) >> 1) > buflen) {
+	if (hex_decode_l(len) > buflen) {
 		errno = ENOMEM;
 		return (-1);
 	}
+
+	/* XXX: prepended extra nibble, not appended! */
```

Comment:
Of course. We expect "3ab" to be decoded as 0x03ab, not 0x3ab0. This is documented, and changing it would lead to regressions, fail current VTC tests, and IMO would violate common intuitions.